sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -920,9 +921,12 @@ object DDLUtils {
             serde == Some("parquet.hive.serde.ParquetHiveSerDe") ||
             serde == Some("org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe")) {
             ParquetSchemaConverter.checkFieldNames(colNames)
+          } else if (serde == HiveSerDe.sourceToSerDe("avro").get.serde) {
+            AvroFileFormat.checkFieldNames(colNames)
           }
         case "parquet" => ParquetSchemaConverter.checkFieldNames(colNames)
         case "orc" => OrcFileFormat.checkFieldNames(colNames)
+        case "avro" => AvroFileFormat.checkFieldNames(colNames)
```

Comment:
What we can do here is:
```
DataSource.lookupDataSource(provider, conf) match  {
  case f: FileFormat => f.checkFieldNames... // a new API
  case _ =>
}
```

Reply:
then for hive provider part,  we use similar way?
```
} else if (serde == HiveSerDe.sourceToSerDe("avro").get.serde) {
            DataSource.lookupdataSource("avro", conf).checkFieldNames(colNames)
}
```

Reply:
Updated as your comment, but many test failed since 
```
 org.apache.spark.sql.AnalysisException: Failed to find data source: avro. Avro is built-in but external data source module since Spark 2.4. Please deploy the application as per the deployment section of "Apache Avro Data Source Guide".
```
Then move all these test case to Avro module?

Reply:
Any suggestion about this ?